modules/ting_covers_rest/src/TingCoversRestService.inc

```diff
@@ -0,0 +1,120 @@
+<?php
+
+use GuzzleHttp\Client;
+use OpenAPI\Client\Configuration;
+use OpenAPI\Client\Api\DefaultApi;
+
+/**
+ * @file
+ * High level implementation of the Ting Covers REST API.
+ */
+class TingCoversRestService {
+  private $config;
+  private $api;
+
+  /**
+   * Construct an API instance.
+   *
+   * @param $host
+   *    The REST service hostname.
+   *
+   * @param bool $token
+   *    The OAuth2 token.
+   *
+   * @param bool $debug
+   *    Enable/disable debugging.
+   */
+  public function __construct($host, $token = FALSE, $debug = FALSE) {
+    $this->config = new Configuration();
+    $this->config->setHost($host)->setDebug($debug);
+    $token ?: $this->config->setAccessToken($token);
```

Comment:
Ah... den syntax er da rigtigt nok men her synes jeg nu ikke om at short hande sig væk fra en almindelig `if`.

Don't make me think. Jeg måtte lige stoppe op og tænke over hvad der sker og om det er korrekt -- det er helt unødvendigt.

Reply:
Ikke enig, men ok. ;)

Reply:
Ironisk nok gør OpenAPI generatoren det samme. 😄

```
    public function __construct(
        ClientInterface $client = null,
        Configuration $config = null,
        HeaderSelector $selector = null
    ) {
        $this->client = $client ?: new Client();
        $this->config = $config ?: new Configuration();
        $this->headerSelector = $selector ?: new HeaderSelector();
    }
```

Reply:
Optimize for readability

Reply:
Men den gør da ikke det samme. Der assigner den værdien til en variable. Det gør du ikke.

Reply:
Det er ikke short hands generelt og altid jeg ikke bryder mig om.

Reply:
Det er rettet